docs/DeveloperGuide.adoc

```diff
@@ -620,6 +620,84 @@ image::HideInfoSequenceDiagrams.png[]
 
 The censored data are only hidden from the window, not changed in the database. The `BrowserPanel` on the bottom right of the window will still show all the information including the hidden information.
 
+=== Send Reminder Email feature
+
+==== Current Implementation
+
+The email sending mechanism is supported by the https://docs.oracle.com/javaee/7/api/javax/mail/package-summary.html[`JavaMail`] library. This feature is composed of three commands: `check`, `setemail` and `remind`.
+
+* `check` command: display the email address that the user has set to send emails from. The app will censor the email address.
+[NOTE]
+If user has not set an email, the email address will be a string `"default"` and the app will throw a `CommandException`.
+
+* `setemail` command: set the email address used for sending reminders to the customers.
+[NOTE]
+Only gmail is accepted by the app.
+
+* `remind` command: automatically generate an email containing some core information about the loan, and send it to the corresponding customer from user's email. This command will use the `JavaMail` library.
+[NOTE]
+The email will contain the `Name`, `BikeId`, `LoanStartTime` and `LoanRate` of the loan.
+
+Given below is an example usage scenario and how the reminder email mechanism behaves at each step.
+
+*Step 1.* The user launches the application for the first time. The user's email address is string `"default"` by default.
+
+*Step 2.* The user executes `check`. The command executes `ModelManager#getMyEmail()`, which calls `UserPrefs#getDefaultEmail()` and returns user's email stored in `UserPref`. The app checks if user's email equals to `"default"`. In this case, they are equal, so it throws a `CommandException`.
```

Comment:
Could use a sequence diagram.

Reply:
Do you mean a sequence diagram to show the whole procedure or just this step?